test/integration/test-promise-wrappers.js

```diff
@@ -52,8 +52,31 @@ function testErrors () {
   });
 }
 
+function testObjParams () {
+  var connResolved;
+  var connPromise = createConnection(config).then(function (conn) {
```

Comment:
To prevent regressions it would be nice if you could test this against the pool aswell (I do this in my tests aswell with a simple loop)